x-pack/plugins/alerting/server/types.ts

```diff
@@ -76,6 +77,8 @@ export interface AlertServices<
     id: string
   ) => PublicAlertInstance<InstanceState, InstanceContext, ActionGroupIds>;
   shouldWriteAlerts: () => boolean;
+  shouldStopExecution: () => boolean;
+  search: IAbortableClusterClient;
```

Comment:
so this means that we should use services.search [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts#L73) as that's the abortable cluster client & good for cancellation when we start using the feature

Reply:
Yes that's correct! If you have an ES query that happens to be running 5 minutes into a rule execution (5 minutes is the default rule task timeout unless you have overridden it for detection rules, the query will abort with an error. You will want to wrap that call in a try/catch if you want to add your own error handling.